### DISCUSSION TITLE
Why is the GitHub Action erroring?

### DISCUSSION THREAD

**Discussion by Ombucha** [lowlighter/metrics#1046] [Q&A] [answered]
**Link:** https://github.com/Pysics/Pysics/actions/runs/2299900663

**Comment by lowlighter** [comment, marked as answer]
Hi 👋 !

Seems the github token does not have write access to target repository:
![image](https://user-images.githubusercontent.com/22963968/168125758-2701b492-e086-4440-9d2b-4ee9ffabf11b.png)

So when it tries to commit the result, it returns a 403 error (forbidden).
[`comitter_token`](https://github.com/lowlighter/metrics/blob/master/source/plugins/core/README.md#committer_token) (which default to `${{ github.token }}`) needs write permissions to push changes

**Reply by rvalitov** [reply]
The token permissions are configured in the settings of the repository:
![image](https://user-images.githubusercontent.com/16267156/168719449-305e5307-9b16-4307-8ecd-7560c46d2542.png)

By default the permissions are read only.

**Reply by lowlighter** [reply]
Yes, it's also possible to set it at worfflow/job level:

```yml
metrics:
  permissions:
    contents: write
  steps:
    - uses: lowlighter/metrics@latest
```
